core/templates/pages/topic-editor-page/editor-tab/topic-editor-tab.directive.spec.ts

```diff
@@ -648,4 +648,39 @@ describe('Topic editor tab directive', function() {
     topicReinitializedEventEmitter.emit();
     expect(ctrl.initEditor).toHaveBeenCalledTimes(2);
   });
+
+  it('should call the TopicUpdateService if skillId is added in the ' +
+     'diagnostic test', function() {
+    var updateSkillIdForDiagosticTestSpy = spyOn(
+      TopicUpdateService, 'updateDiagnosticTestSkills');
+    $scope.addSkillForDiagnosticTest(skillSummary);
+    expect(updateSkillIdForDiagosticTestSpy).toHaveBeenCalled();
+  });
+
+  it('should call the TopicUpdateService if any skillId is removed from the ' +
+     'diagnostic test', function() {
+    var updateSkillIdForDiagosticTestSpy = spyOn(
+      TopicUpdateService, 'updateDiagnosticTestSkills');
+    $scope.selectedSkillsForDiagnosticTest = [skillSummary];
+
+    $scope.removeSkillFromDiagnosticTest(skillSummary);
+    expect(updateSkillIdForDiagosticTestSpy).toHaveBeenCalled();
+  });
+
+  it('should be able to present diagnostic test dropdown selector correctly',
+    function() {
+      $scope.presentDiagnosticTestSkillDropdown();
+      expect($scope.isDiagnosticTestSkillsDropdownPresent).toBeTrue();
+
+      $scope.removeSkillDropdownForDiagnosticTest();
+      expect($scope.isDiagnosticTestSkillsDropdownPresent).toBeFalse();
+    });
+
+  it('should not call the TopicUpdateService if null skill is added in the' +
+     ' diagnostic test', function() {
+    var updateSkillIdForDiagosticTestSpy = spyOn(
+      TopicUpdateService, 'updateDiagnosticTestSkills');
+    $scope.addSkillForDiagnosticTest(null);
```

Comment:
Shouldn't this throw some kind of error?

Reply:
This comment is linked to another comment: https://github.com/oppia/oppia/pull/15742/files#r925988626